File: 25_02_2025_ Number_of_Sub-arrays_With_Odd_Sum.py
class Solution(object):
    def numOfSubarrays(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        oddcount = 0
        prefixsum = 0
        arrSize = len(arr)
        
        for num in arr:
            prefixsum += num
            oddcount += prefixsum % 2
        
        oddcount = oddcount + (arrSize - oddcount) * oddcount
        return oddcount % 1000000007

File: 25_02_2025_ Number_of_Sub-arrays_With_Odd_Sum.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr) {
        long long oddcount = 0, prefixsum = 0;
        int arrSize = arr.size();
        
        for (int num : arr) {
            prefixsum += num;
            oddcount += prefixsum % 2;
        }
        
        oddcount = oddcount + (arrSize - oddcount) * oddcount;
        return oddcount % 1000000007;

    }
};
